flow-server/src/main/java/com/vaadin/flow/server/frontend/scanner/FrontendDependencies.java

```diff
@@ -637,7 +644,15 @@ private void computePackages() throws ClassNotFoundException, IOException {
                         "Multiple npm versions for {} found:  {}. First version found '{}' will be considered.",
                         dependency, foundVersions, version);
             }
-            packages.put(dependency, version);
+            Set<Boolean> devs = npmPackageVisitor.getValuesForKey(VALUE,
+                    dependency, DEV);
+            devs.forEach(dev -> {
+                if (dev) {
```

Comment:
The getValuesForKey only finds given values, so it doesn't check for the default value.
Either we should add a null check for dev here which would equal to false:
```suggestion
                if (dev != null && dev) {
```
Or then the getValuesForKey method in `FrontendAnnotatedClassVisitor` should have the map changed to `.map(h -> h.get(property) == null ? getDefault(property): h.get(property))` so that we would get the correct value.

Reply:
Seems that accepting defaults for the valuesForKey generate probelms in other places so the check should just have a null be the same as `dev=false`